sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -233,6 +233,148 @@ public virtual Response<T> CreateEntity<T>(T entity, CancellationToken cancellat
+        /// <exception cref="ArgumentNullException"><paramref name="partitionKey"/> or <paramref name="rowKey"/> is null.</exception>
+        public virtual async Task<Response<T>> GetEntityAsync<T>(string partitionKey, string rowKey, CancellationToken cancellationToken = default) where T : TableEntity, new()
+        {
+            Argument.AssertNotNull("message", nameof(partitionKey));
+            Argument.AssertNotNull("message", nameof(rowKey));
+
+            using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TableClient)}.{nameof(GetEntity)}");
+            scope.Start();
+            try
+            {
+                var response = await _tableOperations.QueryEntitiesWithPartitionAndRowKeyAsync(
+                    _table,
+                    partitionKey,
+                    rowKey,
+                    queryOptions: new QueryOptions() { Format = _format },
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                var result = ((Dictionary<string, object>)response.Value).ToTableEntity<T>();
+                return Response.FromValue(result, response.GetRawResponse());
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets the specified table entity.
+        /// </summary>
+        /// <param name="partitionKey">The partitionKey that identifies the table entity.</param>
+        /// <param name="rowKey">The rowKey that identifies the table entity.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>The <see cref="Response"/> indicating the result of the operation.</returns>
+        /// <exception cref="RequestFailedException">Exception thrown if entity already exists.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="partitionKey"/> or <paramref name="rowKey"/> is null.</exception>
+        public virtual Response<IDictionary<string, object>> GetEntity(string partitionKey, string rowKey, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull("message", nameof(partitionKey));
+            Argument.AssertNotNull("message", nameof(rowKey));
+
+            using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(TableClient)}.{nameof(GetEntity)}");
+            scope.Start();
+            try
+            {
+                var response = _tableOperations.QueryEntitiesWithPartitionAndRowKey(
+                    _table,
+                    partitionKey,
+                    rowKey,
+                    queryOptions: new QueryOptions() { Format = _format },
+                    cancellationToken: cancellationToken);
+
+                IDictionary<string, object> entity = (IDictionary<string, object>)response.Value;
+                entity.CastAndRemoveAnnotations();
```

Comment:
Would customers ever want these annotations in niche scenarios?  It might be interesting to give the raw values back if customers pass a flag or something.

Reply:
I can't think of a good reason to expose them, since they are mostly a protocol implementation detail. We could always add it as an option later, maybe as part of `TableClientOptions` if feedback guides us that way.

Reply:
Sounds good.  This is one to pick Jeff's brain about during the next Arch Board review.